tokio-tcp/src/stream.rs

```diff
@@ -784,8 +784,11 @@ impl TcpStream {
     #[deprecated(since = "0.1.14", note = "use `split()` instead")]
     #[doc(hidden)]
     pub fn try_clone(&self) -> io::Result<TcpStream> {
-        let io = self.io.get_ref().try_clone()?;
-        Ok(TcpStream::new(io))
+        // Rationale for deprecation:
+        // - https://github.com/tokio-rs/tokio/pull/824
+        // - https://github.com/tokio-rs/tokio/issues/774#issuecomment-451059317
+        let msg = "`TcpStream::split()` is deprecated because it doesn't work as intended";
```

Comment:
I think the message should be `TcpStream::try_clone()` instead of `TcpStream::split()`

Reply:
This was eventually fixed: https://github.com/tokio-rs/tokio/blob/847fb59b17d8e9786abfdac474ad59e120a94254/tokio-tcp/src/stream.rs#L788